src/fsd/1-pages/learn-characters/rank-lookup.service.ts

```
@@ -0,0 +1,225 @@
+
+        return result;
+    }
+
+    public static getAllMaterials(
+        campaingsProgress: ICampaignsProgress,
+        ownedUpgrades: Record<string, number>,
+        upgrades: IMaterialFull[]
+    ): IMaterialEstimated2[] {
+        const result = UpgradesService.groupBaseMaterials(upgrades)
+            .map(x =>
+                this.calculateMaterialData(
```

Biome (1.9.4)

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)